src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -59,7 +56,7 @@ public class Program
         static void Main(string[] args)
         {
             var a = new Fraction();
-            var b = [|+a|];
+            var b = {|CA2252:+a|};
```

Comment:
Are the different rules ids for distinguishing diagnostics?  You can verify with the diagnostic descriptor instead https://github.com/dotnet/roslyn-analyzers/blob/682876f0e2ecfce5692c90eb015831d7cd863610/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidZeroLengthArrayAllocationsTests.cs#L118

Reply:
Yup I added the new ruleIDs so I could use the easy markup syntax for the unit tests. Multiple diagnostics defined in the same rule ID didn't seem to work well with the markup syntax.

Do you feel strongly about having only 1 diagnostic ID here? Is it wrong to use multiple diagnostic IDs in 1 analyzer? If I have to change to using `VerifyCS.Diagnostic`, I have to modify every unit test with the row and column numbers. Seems like a waste of time?

Reply:
> Is it wrong to use multiple diagnostic IDs in 1 analyzer?

Is there a reason a developer would want to configure them independently, e.g. different warning levels or some enabled and some not?  If not, we should have only one.  We shouldn't add additional IDs just for testing purposes.

Reply:
> Is there a reason a developer would want to configure them independently

I can't think of a natural reason for this. I'll change to using `VerifyCS.Diagnostic`